packages/core/src/components/MonthPicker/MonthPicker.jsx

```diff
@@ -0,0 +1,220 @@
+ * language of the locale. Full month names are also included as `aria-label`
+ * attributes.
+ */
+export class MonthPicker extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.months = getMonthNames(props.locale);
+    this.monthsLong = getMonthNames(props.locale, false);
+
+    if (typeof this.props.selectedMonths === 'undefined') {
+      this.isControlled = false;
+      // Since this isn't a controlled component, we need a way
+      // to track when the value has changed.
+      this.state = {
+        selectedMonths:
+          this.props.defaultSelectedMonths || this.monthsArray(false)
+      };
+    } else {
+      this.isControlled = true;
+    }
+  }
+
+  selectedMonths() {
+    if (this.isControlled) {
+      return this.props.selectedMonths;
+    }
+
+    return this.state.selectedMonths;
+  }
+
+  disabledMonths() {
+    return this.props.disabledMonths || this.monthsArray(false);
+  }
+
+  /**
+   * Generates an array of 12 booleans with the given value
+   *
+   * @param  {boolean} [selected] value to populate array items
+   * @return {boolean[]}          array of 12 booleans
+   */
+  monthsArray(selected) {
+    const array = [];
+    for (let i = 0; i < NUM_MONTHS; i++) {
+      array[i] = selected;
+    }
+    return array;
+  }
+
+  handleChange(event) {
+    if (this.props.onChange) {
+      this.props.onChange(event);
+    }
+
+    if (!this.isControlled) {
+      const index = event.target.value;
+      const selectedMonths = this.state.selectedMonths.slice();
+      selectedMonths[index] = !selectedMonths[index];
+      this.setState({ selectedMonths });
+    }
+  }
+
+  handleSelectAll() {
+    if (this.props.onSelectAll) {
+      this.props.onSelectAll();
+    }
+
+    if (!this.isControlled) {
+      this.setState({
+        selectedMonths: this.disabledMonths().map(disabled => !disabled)
+      });
+    }
+  }
+
+  handleClearAll() {
+    if (this.props.onClearAll) {
+      this.props.onClearAll();
+    }
+
+    if (!this.isControlled) {
+      this.setState({ selectedMonths: this.monthsArray(false) });
+    }
+  }
+
+  renderMonths(selectedMonths) {
+    const disabledMonths = this.disabledMonths();
+    const { name, inversed } = this.props;
+    return (
+      <ul className="ds-c-list--bare ds-u-display--flex ds-u-justify-content--between ds-u-flex-wrap--wrap">
+        {this.months.map((month, i) => (
+          <Choice
+            name={name}
+            value={i}
+            checked={selectedMonths[i]}
+            key={month}
+            onChange={e => this.handleChange(e)}
+            className="ds-c-month-picker__month"
+            disabled={disabledMonths[i]}
+            inversed={inversed}
+            aria-label={this.monthsLong[i]}
+          >
+            {month}
+          </Choice>
+        ))}
+      </ul>
+    );
+  }
+
+  renderButton(text, onClick) {
+    return (
+      <Button
+        variation="primary"
```

Comment:
I'd push for keeping this as the default button style, at least by default. We could add support for setting what button variation gets used (ie. adding a `selectAllButtonVariation` and `clearAllButtonVariation` props). In general, it's frowned upon to have multiple primary call-to-actions on the same page and I'd like to avoid encouraging it. (For HealthCare.gov we actually use the "success" variation as our primary CTA, so I understand why this is like it is).

Here's with the default button style looks like, which I think I actually prefer since these actions don't seem like they deserve as much focus:

![image](https://user-images.githubusercontent.com/371943/33344964-ade8ea0e-d458-11e7-90ba-1544b00141ab.png)

Reply:
@cindyphan what do you think about this visually?

Reply:
To clarify: The blue button style would still be possible if we want to do that on HealthCare.gov (using the props I suggest above), however what I'm advocating for here is to not set these buttons as primary buttons by default because sites implementing the design system may actually be using the blue button as their primary button style (since that's the standard).

Reply:
My confusion on this arose from the fact that the solid blue is the secondary color in HC.gov. They're not primary anymore, and I've added an optional `buttonVariation` prop.